test/functional/lua/loop_spec.lua

```diff
@@ -0,0 +1,55 @@
+-- Test suite for testing interactions with API bindings
+local helpers = require('test.functional.helpers')(after_each)
+local funcs = helpers.funcs
+local meths = helpers.meths
+local clear = helpers.clear
+local sleep = helpers.sleep
+local eq = helpers.eq
+local matches = helpers.matches
+
+before_each(clear)
+
+describe('vim.loop', function()
+
+  it('version', function()
+    assert(funcs.luaeval('vim.loop.version()')>=72961, "libuv version too old")
+    matches("(%d+)%.(%d+)%.(%d+)", funcs.luaeval('vim.loop.version_string()'))
+  end)
+
+  it('timer', function()
+    meths.execute_lua('vim.api.nvim_set_var("coroutine_cnt", 0)', {})
+
+    local code=[[
+      local loop = vim.loop
+
+      local touch = 0
+      local function wait(ms)
+        local this = coroutine.running()
+        assert(this)
+        local timer = loop.new_timer()
+        timer:start(ms, 0, function ()
+          timer:close()
+          touch = touch + 1
+          coroutine.resume(this)
+          touch = touch + 1
+          assert(touch==3)
+          vim.api.nvim_set_var("coroutine_cnt_1", touch)
+        end)
+        coroutine.yield()
+        touch = touch + 1
+        return touch
+      end
+      coroutine.wrap(function()
+        local touched = wait(100)
+        assert(touched==touch)
+        vim.api.nvim_set_var("coroutine_cnt", touched)
+      end)()
+    ]]
+
+    eq(0, meths.get_var('coroutine_cnt'))
+    meths.execute_lua(code, {})
+    sleep(200)
```

Comment:
does it work if this is `20` and we do `wait(10)`? We want the tests to be fast, if possible.

Reply:
let me have a try.